File: c/medium/minimum_deletions_to_make_string_balanced.c
#include <stdio.h>

int minimumDeletions(char* s) {
    int min_deletions = 0;
    int b_before_a = 0;

    for (int i = 0; s[i] != '\0'; i++) {
        if (s[i] == 'b') {
            b_before_a++;
        } else if (b_before_a > 0) {
            min_deletions++;
            b_before_a--;
        }
    }

    return min_deletions;
}

File: rust/medium/minimum_deletions_to_make_string_balanced.rs
pub fn minimum_deletions(s: &str) -> i32 {
    let mut min_deletions = 0;
    let mut b_before_a = 0;

    for c in s.chars() {
        if c == 'b' {
            b_before_a += 1;
        } else if b_before_a > 0 {
            min_deletions += 1;
            b_before_a -= 1;
        }
    }

    min_deletions
}
